git_archive_all.py

```diff
@@ -227,38 +170,14 @@ def is_file_excluded(self, repo_abspath, repo_file_path, exclude_patterns):
         @param repo_file_path: Path to a file within repo_abspath.
         @type repo_file_path: str
 
-        @param exclude_patterns: Exclude patterns with format specified for get_exclude_patterns.
-        @type exclude_patterns: dict
-
         @return: True if file should be excluded. Otherwise False.
         @rtype: bool
         """
-        if exclude_patterns is None or not len(exclude_patterns):
-            return False
-
-        from fnmatch import fnmatch
-
-        file_name = path.basename(repo_file_path)
-        components = self.get_path_components(repo_abspath, path.join(repo_abspath, path.dirname(repo_file_path)))
-
-        is_excluded = False
-        # We should check all patterns specified in intermediate directories to the given file.
-        # At the end we should also check for the global patterns (key '()' or empty tuple).
-        while not is_excluded:
-            key = tuple(components)
-            if key in exclude_patterns:
-                patterns = exclude_patterns[key]
-                for p in patterns:
-                    if fnmatch(file_name, p) or fnmatch(repo_file_path, p):
-                        self.LOG.debug("Exclude pattern matched {0}: {1}".format(p, repo_file_path))
-                        is_excluded = True
-
-            if not len(components):
-                break
-
-            components.pop()
-
-        return is_excluded
+        out = self.run_git_shell(
+            'git check-attr -a -- %s' % repo_file_path,
```

Comment:
@legoktm Any reason why `--all` was preferred over a specific attribute? As far as I understand `git check-attr export-ignore` should have worked.

Also I think we should use `-z` and split by `NUL` instead of `in`.